### PR TITLE
Order rollback

### PR DESCRIPTION
**Pull request by yorickdevries**
Rolls back orders if payment or stock is insufficient

### CODE REVIEW

**Comment by yorickdevries** [comment]
@qurben Can you look into the failed CI ?

**Comment by qurben** [comment]
Weird flaky behaviour 🤷‍♂. Might be nice to be a bit more verbose with the error messages here.

File: common/src/main/java/nl/tudelft/wdm/group1/common/OrderStatus.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package nl.tudelft.wdm.group1.common;
+
+public enum OrderStatus {
+    PROCESSING,
+    FAILEDDUETOLACKOFSTOCK,
```
Comment (qurben) [inline common/src/main/java/nl/tudelft/wdm/group1/common/OrderStatus.java]: Please make readable using underscores.

File: service/stock/src/main/java/nl/tudelft/wdm/group1/stock/events/Consumer.java [hunk 2]
```diff
@@ -77,4 +77,22 @@ public void consumeOrderCheckedOut(final Order order)
...
+        // Only perform action when the order was cancelled due to lack of payment
+        if(order.getStatus() == OrderStatus.FAILEDDUETOLACKOFPAYMENT){
+            // TODO: lock the stock while adding
+            for (UUID stockItemId : order.getItemIds()) {
+                try {
+                    StockItem stockItem = stockItemRepository.findOrElseThrow(stockItemId);
+                    stockItem.addStock(1);
+                } catch (ResourceNotFoundException | InvalidStockChangeException e) {
+                    // we assume this should not cause any exception
+                    e.printStackTrace();
```
Comment (qurben) [inline service/stock/src/main/java/nl/tudelft/wdm/group1/stock/events/Consumer.java]: Use the logger if you want to print something.

**Pull request merged**